### DISCUSSION TITLE
Search Query Enhancer in V4

### DISCUSSION THREAD

**Discussion by lucadalsacco** [microsoft-search/pnp-modern-search#2142] [Q&A]
Hello,
I have a Sharepoint site where we installed the v4 of your extension.
Now I need to extend the search to apply some enhancement to query (also using LUIS) and to record the query text in a database.
I saw that in V3 there is the Search Query Enhancer extension that perfectly meet my needs.
So my questions are:
1) Is the same functionality available in V4? If yes, which extension can I use?
2) If not, can you tell me when the Search Query Enhancer will be available in V4?
3) Is there any way to install a V3 extension (e.g. the Search Query Enhancer) in V4 search?

Thank you very much!

**Comment by FranckyC** [comment]
Hi @lucadalsacco, did you take a look at the SPFx Search Query extension capability ? https://docs.microsoft.com/en-us/sharepoint/dev/spfx/building-search-extensions

**Reply by lucadalsacco** [reply]
Hi @FranckyC. I took a look at it. I don't understand if that extension works with the Modern Search Query Box or it works only with the standard search query box. Furthermore it seems to be not installable in Share Point Online.
